nntrainer/tensor/manager.cpp

```diff
@@ -250,6 +255,7 @@ void Manager::initializeWeights() {
     for (auto &w : l_w) {
       Weight &weight = w.get();
       auto dim = weight.getDim();
+      /** This will allocate memory for weights right now */
```

Comment:
we do allocation and initialization weight here? or do we call allocateWeights first before initialization? If it is second, then we need to check the weights_allocated before initialization.

Reply:
> we do allocation and initialization weight here? 

Yes, weights initialization and allocation are done together here.

The check is missing - added a check of weights_initialized here.